allennlp/tango/dataset.py

```diff
@@ -0,0 +1,68 @@
+"""
+*AllenNLP Tango is an experimental API and parts of it might change or disappear
+every time we release a new version.*
+"""
+
+import itertools
+from dataclasses import dataclass, field
+from typing import Mapping, Any, Optional, Sequence, Dict
+
+from allennlp.data import Vocabulary, DatasetReader, Instance
+from allennlp.tango.step import Step
+from tqdm import tqdm
+
+
+@dataclass
+class DatasetDict:
```

Comment:
If we're going to call this `DatasetDict` (which I like), I think this should act like a `dict`. At least we should implement `__getitem__` so you can do `dataset_dict[split_name]`.

Reply:
I put in `__getitem__` and `__len__`, but I am actually not a big fan. I want people to be at least aware that those other members are there, and making it act more and more like a dict is putting in a lot of magic that might bite us later.